Whoaverse/Whoaverse/Scripts/whoaverse.ui.js

```diff
@@ -809,15 +820,44 @@ VimeoExpando.prototype.process = function (target) {
 };
 
 /* SoundCloud */
-var SoundCloudExpando = function () {
-    var clientId = 'ab19f68dc1985a1b24752d987c91b7aa';
-    IFrameEmbedderExpando.call(this, /xxx/i);
+var SoundCloudExpando = function (options) {
+    
+    IFrameEmbedderExpando.call(this, /http(?:s{0,1}):\/\/(?:soundcloud\.com|snd\.sc)\/(.*)\/?/i, options);
+    this.getSrcUrl = function (id, fn, fnError) {
+
+        this.getSourceInfo = function (id, fnCallback, fnErrorHandler) {
+            $.getJSON(UI.Common.currentProtocol() + '//soundcloud.com/oembed', {
+                format: 'json',
+                url: 'https://soundcloud.com/' + id,
+                iframe: 'true'
+            }).done(function (json) {
+                if (!json.html) {
+                    fnErrorHandler();
+                    return;
+                }
+                var regex = /<iframe.*?src="(.*?)"/;
```

Comment:
I think it is useful to comment regular expressions for people who are not familiar.